src/sinks/prometheus/remote_write/sink.rs

```diff
@@ -0,0 +1,235 @@
+use std::fmt;
+
+use vector_common::byte_size_of::ByteSizeOf;
+use vector_core::{
+    event::Metric,
+    stream::batcher::{data::BatchData, limiter::ByteSizeOfItemSize},
+};
+
+use crate::sinks::{prelude::*, util::buffer::metrics::MetricSet};
+
+use super::{
+    request_builder::{RemoteWriteEncoder, RemoteWriteRequest, RemoteWriteRequestBuilder},
+    PartitionKey, PrometheusMetricNormalize,
+};
+
+pub(super) struct RemoteWriteMetric {
+    pub(super) metric: Metric,
+    tenant_id: Option<String>,
+}
+
+impl Finalizable for RemoteWriteMetric {
+    fn take_finalizers(&mut self) -> EventFinalizers {
+        self.metric.take_finalizers()
+    }
+}
+
+impl GetEventCountTags for RemoteWriteMetric {
+    fn get_tags(&self) -> TaggedEventsSent {
+        self.metric.get_tags()
+    }
+}
+
+impl EstimatedJsonEncodedSizeOf for RemoteWriteMetric {
+    fn estimated_json_encoded_size_of(&self) -> JsonSize {
+        self.metric.estimated_json_encoded_size_of()
+    }
+}
+
+impl ByteSizeOf for RemoteWriteMetric {
+    fn allocated_bytes(&self) -> usize {
+        self.metric.allocated_bytes()
+    }
+}
+
+#[derive(Clone, Copy, Debug, Default)]
+pub struct PrometheusRemoteWriteDefaultBatchSettings;
+
+pub(super) struct PrometheusTenantIdPartitioner;
+
+impl Partitioner for PrometheusTenantIdPartitioner {
+    type Item = RemoteWriteMetric;
+    type Key = PartitionKey;
+
+    fn partition(&self, item: &Self::Item) -> Self::Key {
+        PartitionKey {
+            tenant_id: item.tenant_id.clone(),
+        }
+    }
+}
+
+impl SinkBatchSettings for PrometheusRemoteWriteDefaultBatchSettings {
+    const MAX_EVENTS: Option<usize> = Some(1_000);
+    const MAX_BYTES: Option<usize> = None;
+    const TIMEOUT_SECS: f64 = 1.0;
+}
```

Comment:
Ordering nit - can we move `impl SinkBatchSettings for PrometheusRemoteWriteDefaultBatchSettings` up so that it is right after line 46?